### DISCUSSION TITLE
MobX React replacement for componentWillReact() for triggering layout animations

### DISCUSSION THREAD

**Discussion by jeremy303** [mobxjs/mobx#2864] [Q&A]
It was once [recommended to use `componentWillReact()` to trigger layout animations](https://github.com/mobxjs/mobx-react/issues/332) in response to changes in observable values, like so:

```
componentWillReact() {
        LayoutAnimation.configureNext(LayoutAnimation.Presets.spring);
    }
```

However, `componentWillReact()` has been removed in mobx-react@6.

What is the recommended replacement?  My _observable_ class should know nothing about it's _observer_, so I'm looking for a similar event-driven solution.

(Full disclosure: [I asked this question on Stack Overflow](https://stackoverflow.com/questions/63820236/mobx-react-replacement-for-componentwillreact-for-triggering-layout-animations) about 6 months ago but have received no responses)

Thanks!

**Comment by urugator** [comment]
`componentDidUpdate` should work fine I quess
